src/main/java/org/candlepin/guice/CandlepinContextListener.java

```diff
@@ -426,17 +450,46 @@ protected void checkDbChangelog() {
                         .append(changeset.getFilePath())
                         .append('\n');
                 });
-
-                log.error(builder.toString());
-                log.error("Please update the database");
-                log.error("Aborting Candlepin initialization...");
-
-                throw new RuntimeException("The database is missing Liquibase changeset(s)");
+                // all levels above None need this
+                log.warn(builder.toString());
+
+                if (liquibaseOnStart.equals(DBManageLevel.REPORT.getName())) {
+                    log.warn("Please update the database");
+                }
+                else if (liquibaseOnStart.equals(DBManageLevel.HALT.getName())) {
+                    log.error("Aborting Candlepin initialization...");
+                    throw new RuntimeException("The database is missing Liquibase changeset(s)");
+                }
+                else if (liquibaseOnStart.equals(DBManageLevel.MANAGE.getName())) {
+                    log.warn("Calling liquibase to update the database");
+                    liquibase.update();
+                    log.warn("Update complete");
+                    liquibase.forceReleaseLocks();
+                }
+                else {
+                    log.error("The value for 'candlepin.db.database_manage_on_startup' is invalid. " +
+                        "No action taken.");
+                }
             }
         }
         catch (LiquibaseException e) {
+            log.error("Liquibase exception: " +  e.getMessage());
             throw new RuntimeException(e);
         }
+        finally {
+            if (liquibase != null && liquibase.getDatabase() != null) {
+                DatabaseConnection dc = liquibase.getDatabase().getConnection();
+                if (dc != null) {
+                    try {
+                        dc.rollback();
```

Comment:
Do we always want to roll back in this case?

Reply:
I can see keeping sets that ran successfully before an error.